### PR TITLE
Reject functions without bytecode from `_can_be_jitted` in GroupBy Apply

### PR DESCRIPTION
**Pull request by brandon-b-miller**
Closes https://github.com/rapidsai/cudf/issues/13426

### CODE REVIEW

**Review by bdice — COMMENTED** [review]
Can we add a test for the case in the issue this closes? We should not allow a regression that breaks past behavior even if it falls back to the old implementation.

File: python/cudf/cudf/core/udf/groupby_utils.py [hunk 1]
```diff
@@ -209,6 +209,9 @@ def _can_be_jitted(frame, func, args):
     by attempting to compile just the function to PTX using the
     target set of types
     """
+    if not hasattr(func, "__code__"):
+        # numba requires bytecode to be present to proceed
```
Comment (bdice) [inline python/cudf/cudf/core/udf/groupby_utils.py:213]:
```suggestion
        # Numba requires bytecode to be present to proceed.
        # See https://github.com/numba/numba/issues/4587
```

**Comment by brandon-b-miller** [comment]
> Can we add a test for the case in the issue this closes? We should not allow a regression that breaks past behavior even if it falls back to the old implementation.

What do you think of the test added in https://github.com/rapidsai/cudf/pull/13429/commits/51633570a2c755a2150786a7304bf572bcee0440 ?

**Review by mroeschke — APPROVED** [review]

**Review by bdice — APPROVED** [review]

**Pull request merged**